src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -1136,6 +1139,16 @@ private void CheckAllowedProjectChanges(ProjectChanges projectChanges)
             }
         }
 
+        private bool CanApplyParseOptionChange(ParseOptions old, ParseOptions @new)
+        {
+            // only changes to C# LanguageVersion are supported at this point
```

Comment:
This shouldn't be happening at the core Workspace type. The description you are defining is a behavior of the host, and should be overridable if necessary.